Concise and readable projectConfig modifications

otterlistic opened the pull request:
Nested if condition was removed and repetition of projectConfig[key] has been avoided.
It's possible to mitigate the usage of forEach and filter on line 11 and 12 using reduce() which will be the next PR

otterlistic committed:
Concise and readable projectConfig modifications

Nested if condition was removed and repetition of projectConfig[key] has been avoided.
It's possible to mitigate the usage of forEach and filter on line 11 and 12 using reduce() which will be the next PR

sthielen commented on src/tools/writeConfig.js:
```diff
@@ -15,16 +15,19 @@ const writeConfig = () => {
+         const fileValue = file[key];
+          if (configValue) {
+            
+            Array.isArray(configValue) && Array.isArray(fileValue))
```
I am getting a parser error here -- possibly requires some babel modifications to use this style of assignment?

sthielen commented on src/tools/writeConfig.js:
```diff
@@ -15,16 +15,19 @@ const writeConfig = () => {
-            // Otherwise, set it
-            projectConfig[key] = file[key];
-          }
+         let configValue = projectConfig[key];
```
Can you check your editorconfig/eslint settings here? Should match indentation style for the rest of the project

sthielen reviewed (changes requested):
Hey! Thanks for putting this together. Added some notes! If it's not too much trouble, want to add the updates using `reduce` to this PR too? I definitely agree that that is the right way forward

otterlistic committed:
Added error logging and edited the PR per comments

reduce() is used instead of filter() and forEach() when creating projectConfig
more concise function parameter naming for creation of serviceMap (line 36-39)
added error catching for generating config.json 

Personally think that the try/catch blocks at line 11 are unnecessary because the path value always exists, therefore line 13 wont fire an error ever. Error catching for scenarios where path is undefined should be handled in the readDirectory() function.

otterlistic commented:
The changes have been made and reduce() has also been included

sthielen commented:
Sweet! LGTM :)
